bugsnag/utils.py

```diff
@@ -233,6 +235,38 @@ def package_version(package_name):
             return None
 
 
+VALIDATION_FAILURE_FORMAT = ('{0} should be {1}, got {2}.' +
+                             ' This will be an error in a future release.')
```

Comment:
I think the only configuration issue that should error is a missing API key; everything else should be a warning

Reply:
Makes sense - I went with API key, endpoints, and delivery, since any of those being incorrect prevents essential usage.